dask/dataframe/io/parquet/arrow.py

```diff
@@ -909,7 +909,11 @@ def _collect_dataset_info(
                     k: hive_categories[k] for k in cat_keys if k in hive_categories
                 }
 
-        if partitioning_supported and ds.partitioning.dictionaries:
+        if (
+            partitioning_supported
+            and ds.partitioning.dictionaries
+            and all(ds.partitioning.dictionaries)
```

Comment:
It might be a little bit more robust to do `all(arr is not None for arr in ds.partitioning.dictionaries)` (currently a pyarrow array evaluates as truthy (and so the above works), but eg a numpy array raises in such a context)